### PR TITLE
separated peakmapper to different file

### PR DESCRIPTION
**Pull request by rdaly525**

### CODE REVIEW

File: metamapper/peakmapper.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        self.peak_primitives = {}
+        self.io_primitives = {}
+        self.discover_constraints = []
+        super(PeakMapper,self).__init__(context,namespace_name)
+ 
+    def extract_instr_map(self,app : coreir.module.Module):
+        instr_map = {}
+        for rr in self.rules:
+            if isinstance(rr,Peak1to1):
+                instr_map = {**instr_map,**rr.instr_map}
```
Comment (rsetaluri) [inline metamapper/peakmapper.py]: `instr_map.update(rr.instr_map)`

File: metamapper/peakmapper.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+
+class PeakMapper(MetaMapper):
+    def __init__(self,context : coreir.context,namespace_name):
+        self.peak_primitives = {}
+        self.io_primitives = {}
+        self.discover_constraints = []
+        super(PeakMapper,self).__init__(context,namespace_name)
+ 
+    def extract_instr_map(self,app : coreir.module.Module):
```
Comment (rsetaluri) [inline metamapper/peakmapper.py:18]: I'm confused. This function doesn't use the argument `app` at all. How is that?

Comment (rdaly525) [reply metamapper/peakmapper.py:18]: Right now I keep around the maps stored in the rewrite rule itself. I want to eventually move into adding the instruction as direct metadata on pycoreir objects, but that is NYI. https://github.com/leonardt/pycoreir/issues/79#issuecomment-479188460

Comment (rsetaluri) [reply metamapper/peakmapper.py:18]: So does this break things if we call `map_app` multiple times on the same mapper object?

Comment (rdaly525) [reply metamapper/peakmapper.py:18]: Yes, that is true. This is a bit of a hack currently.

**Pull request merged**